pkg/kubestatemetrics/builder/builder.go

```diff
@@ -228,7 +228,24 @@ func (b *Builder) startReflector(
 	expectedType interface{},
 	store cache.Store,
 	listWatcher cache.ListerWatcher,
+	useApiServerCache bool,
 ) {
+	if useApiServerCache {
+		listWatcher = &cacheEnabledListerWatcher{ListerWatcher: listWatcher}
+	}
 	reflector := cache.NewReflector(listWatcher, expectedType, store, b.resync*time.Second)
 	go reflector.Run(b.ctx.Done())
 }
+
+type cacheEnabledListerWatcher struct {
+	cache.ListerWatcher
+}
+
+// List uses a resource version of "0" to list all resources.
+// The implementation is required to avoid a quorum from ETCD.
+// https://github.com/kubernetes/kubernetes/blob/dae37c21645c0fc04bbb0202ad1eba6ba4438d23/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L526-L527
```

Comment:
I think the documentation should be linked instead of the code: https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions

Also I'm not sure we should always send `rv = 0`. The way I see it, we should set `rv = 0` and `resourceVersionMatch=NotOlderThan`  the first time we `List` and then `rv = <last observer> and `resourceVersionMatch=NotOlderThan`

Reply:
Thanks! I updated the PR accordingly